src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer.cs

```diff
@@ -305,18 +360,62 @@ bool IsCreateHelper(IMethodSymbol method)
                     method.Parameters[0].Type.SpecialType == SpecialType.System_String;
         }
 
-        private static void AnalyzeTitle(IMethodSymbol descriptorCreationMethod, IFieldInitializerOperation creation, Action<Diagnostic> reportDiagnostic)
+        private static void AnalyzeTitle(OperationAnalysisContext operationAnalysisContext, ImmutableArray<IArgumentOperation> creationArguments, IFieldInitializerOperation creation)
+        {
+            IArgumentOperation titleArgument = creationArguments.FirstOrDefault(a => a.Parameter.Name.Equals("title", StringComparison.OrdinalIgnoreCase));
+
+            if (titleArgument != null &&
+                titleArgument.Parameter.Type != null &&
+                titleArgument.Parameter.Type.SpecialType == SpecialType.System_String)
+            {
+                operationAnalysisContext.ReportDiagnostic(creation.Value.CreateDiagnostic(UseLocalizableStringsInDescriptorRule, WellKnownTypeNames.MicrosoftCodeAnalysisLocalizableString));
+
+                if (titleArgument.Value.ConstantValue.HasValue &&
+                    titleArgument.Value.ConstantValue.Value is string title &&
+                    (title.Contains(".") || ContainsLineReturn(title)))
```

Comment:
Should not fire for cases where '.' is used as a member access, say `Do not use 'A.B' in this context`
```suggestion
                    (title.EndsWith(".") || ContainsLineReturn(title)))
```

Reply:
Doing so we will allow multi-sentences but I guess the check might become complex to ensure that all '.' are wrapped in single-quoted part.

Reply:
We should not fire for `Do not use A.B in this context` either. How about:
```suggestion
                    (title.Contains(". ") || title.EndsWith(".") || ContainsLineReturn(title)))
```